tests/inc/class-test-asset-loader.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * Test methods exported by the base Asset_Loader namespace.
+ */
+
+declare( strict_types=1 );
+
+namespace Asset_Loader\Tests;
+
+use Asset_Loader;
+use WP_Mock;
+
+class Test_Asset_Loader extends Asset_Loader_Test_Case {
+
+	/**
+	 * Registry to mock wp_register/enqueue_script.
+	 *
+	 * @var Mock_Asset_Registry
+	 */
+	private $scripts;
+
+	/**
+	 * Registry to mock wp_register/enqueue_style.
+	 *
+	 * @var Mock_Asset_Registry
+	 */
+	private $styles;
+
+	/**
+	 * String path to production manifest fixture.
+	 *
+	 * @var string
+	 */
+	private $prod_manifest;
+
+	/**
+	 * String path to development manifest fixture.
+	 *
+	 * @var string
+	 */
+	private $dev_manifest;
+
+	public function setUp() : void {
```

Comment:
Some of this method's code could be moved to `setUpBeforeClass`. Pretty everything that is not about setting fields.

Reply:
When I moved the userFunction mocks for the theme environment up, then logic fails -- I don't know why that happens, which is worth exploring (I suspect user function mocks are cleared in WP_Mock's setUp), but since the majority of this logic relates to initializing per-test caches and then binding those caches to user function mocks, most of the logic here to me does seem to need to run before each test.